packages/hash/frontend/src/pages/[account-slug]/types/entity-type/multiple-values-cell.tsx

```diff
@@ -8,109 +8,178 @@ import {
+      sx={{
+        p: "0 !important",
+        position: "relative",
+      }}
+    >
+      <Box
+        onClick={() => {
+          if (multipleValuesMenuOpen) {
+            setMultipleValuesMenuOpen(false);
+          } else {
+            setValue(`properties.${propertyIndex}.array`, true, {
+              shouldDirty: true,
+            });
+            setMultipleValuesMenuOpen(true);
+          }
         }}
-        onClick={() => setMultipleValuesMenuOpen(true)}
+        sx={({ palette }) => ({
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          cursor: "pointer",
+          height: 1,
+          transition: transitions.create("border-color"),
+          border: 1,
+          borderColor: multipleValuesMenuOpen
+            ? `${palette.gray[40]} !important`
+            : "transparent !important",
+        })}
       >
         <Box
-          sx={({ palette, transitions }) => ({
-            display: "flex",
-            justifyContent: "center",
-            alignItems: "center",
-            cursor: "pointer",
-            height: 1,
-            transition: transitions.create("border-color"),
-            border: `1px solid ${
-              multipleValuesMenuOpen ? palette.gray[40] : "transparent"
-            } !important`,
-            "&:hover": {
-              borderColor: `${palette.gray[40]} !important`,
-            },
+          sx={({ palette }) => ({
+            display: "inline-flex",
+            borderRadius: "4px 30px 30px 4px",
+            backgroundColor: "transparent",
+            transition: transitions.create(["padding", "background-color"]),
+            ...(array
+              ? {
+                  py: 0.5,
+                  px: 0.75,
+                  background: palette.gray[20],
+                }
+              : {}),
           })}
         >
-          <Box
-            sx={({ palette, transitions }) => ({
-              display: "inline-flex",
-              borderRadius: "4px 30px 30px 4px",
-              backgroundColor: "transparent",
-              transition: transitions.create(["padding", "background-color"]),
-              ...(array && !multipleValuesMenuOpen
-                ? {
-                    py: 0.5,
-                    px: 0.75,
-                    background: palette.gray[20],
+          <Controller
+            render={({ field: { value, ...field } }) => (
+              <Checkbox
+                {...field}
+                checked={value}
+                onChange={(evt) => {
+                  setMultipleValuesMenuOpen(evt.target.checked);
+                  if (!evt.target.checked) {
+                    setValue(`properties.${propertyIndex}.minValue`, 0, {
+                      shouldDirty: true,
+                    });
+                    setValue(`properties.${propertyIndex}.maxValue`, Infinity, {
+                      shouldDirty: true,
+                    });
```

Comment:
Without the reset, it can be really confusing why the edit bar still displays even if the "on" / "off" state of allow multiple values is back to how it was – because you're still changing the min/max values, they'll just not be applied because array is false.

Reply:
Half-agree with Nate, half with you, Luís.

I **do not** think we should reset the values (i.e. the old behavior is preferable in my mind) while the page is still open. If I accidentally uncheck the box I want to be able to re-check it without having lost my old data inputted.

However I also don't think we should continue to show the 'edit' banner if a user makes a change, and then reverts the change. We should be smart enough to say "it looks again like it used to", meaning that the correct thing to do here would be to persist the array min/max values that are no longer visible in `sessionStorage` alone.

Reply:
Kept the old value is state to restore when you toggle the checkbox. Also changed the way we indicate the maximum value from setting it to infinity to using a flag, so that value isn't overwritten or forgotten either.